src/slicegpt/hf_utils.py

```diff
@@ -74,48 +74,55 @@ def wrapper(*args, **kwargs):
 
 @do_not_initialize
 def get_model_and_tokenizer(
-    model_path: str, uninitialized: bool = False, dtype: torch.dtype = torch.float16, token: str | bool | None = None
+    model_name: str,
+    model_path: str | None = None,
+    uninitialized: bool = False,
+    dtype: torch.dtype = torch.float16,
+    token: str | bool | None = None,
 ) -> tuple[ModelAdapter, PreTrainedTokenizerBase]:
     """Loads the model and the tokenizer from the given path."""
     if uninitialized:
         model_type = "uninitialized"
     else:
         model_type = "pretrained"
 
-    tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=True, token=token)
+    if not model_path:
+        # HF models can be downloaded using the name only, local models need to specify a path
+        model_path = model_name
 
-    logging.info(f"Loading {model_type} {model_path} model")
+    logging.info(f"Loading {model_type} {model_name} model from {model_path}")
```

Comment:
Nit: this is gonna be a bit strange to read if model_name == model_path. Maybe an if/else and `"...{model_name} model from HF"` would be clearer.